cmd/brew-report-issue.rb

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env ruby
+# Creates and closes issues on a project.
+close = !!ARGV.delete("--close")
+user_repo = ARGV.shift
+failure_source = ARGV.shift
+success_meta = failure_source
+
+if user_repo.to_s.empty? || failure_source.to_s.empty?
+  abort "Usage: brew report-issue [--close] <user/repo> <failure-source|success-meta> [<STDIN piped body>]"
+end
+
+unless close
+  abort "Error: the issue's body should be piped over STDIN!" if STDIN.tty?
+  issue_body = STDIN.read
+end
+
+github_credentials=`printf "protocol=https\nhost=github.com\n" | git credential fill 2>/dev/null`
+/username=(?<github_username>.+)/ =~ github_credentials
+/password=(?<github_password>.+)/ =~ github_credentials
+github_username ||= ENV["BOXEN_GITHUB_LOGIN"]
+github_username ||= `git config github.user`.chomp
+
+if github_username.to_s.empty?
+  abort <<-EOS
+Error: your GitHub username is not set! Set it by running Strap:
+  https://strap.githubapp.com
+EOS
+end
+@github_username = github_username
+
+if github_password.to_s.empty?
+  abort <<-EOS
+Error: your GitHub password is not set! Set it by running Strap:
+  https://strap.githubapp.com
```

Comment:
I think this is fine; the repo is mostly public to avoid authentication issues rather than for unmodified usage elsewhere.